node/resolvers/checkout/index.ts

```diff
@@ -152,13 +153,73 @@ export const fieldResolvers = {
 const replaceDomain = (host: string) => (cookie: string) =>
   cookie.replace(/domain=.+?(;|$)/, `domain=${host};`)
 
+const getSession = (ctx: Context) => {
+  return sessionQueries.getSession({}, {}, ctx).catch(err => {
+    // todo: log error using colossus
+    console.error(err)
+    return {} as SessionFields
+  }) as SessionFields
+}
+
+const shouldUpdateClientPreferencesData = (
+  orderFormClientPreferencesData: OrderFormClientPreferencesData,
+  sessionData: SessionFields,
+) => {
+  const sesssionLocale = path(['store', 'cultureInfo'], sessionData)
+  const locale = orderFormClientPreferencesData ? orderFormClientPreferencesData.locale : undefined
+
+  // OrderForm locale should be updated
+  if (sesssionLocale && locale && sesssionLocale !== locale) {
+    return true
+  }
+
+  return false
+}
+
+async function syncWithStoreLocale(
+  orderForm: OrderForm,
+  sessionData: SessionFields,
+  checkout: Checkout
```

Comment:
I dont like to import the Checkout Class just for this type. You can do `checkout['clients']['checkout']` and achieve the same result without the import 😬